synapse/handlers/register.py

```diff
@@ -236,8 +245,10 @@ async def register_user(
         else:
             # autogen a sequential user ID
             fail_count = 0
-            user = None
-            while not user:
+            # If a default display name is not given, generate one.
+            generate_display_name = default_display_name is None
```

Comment:
This fixes a bug where if this went multiple loops only the initial loop would update the display name, which seems incorrect.